### PR TITLE
add OptVendorClass

### PR DESCRIPTION
**Pull request by pmazzini**
* add vendor class option
* reorder the parsing of options

### CODE REVIEW

**Comment by pmazzini** [comment]
The reordering of the switch case when parsing options is what detects as decreasing the coverage. That is why one of the codecov check fails.

File: dhcpv6/option_vendorclass.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// OptVendorClass represents a DHCPv6 Vendor Class option
+type OptVendorClass struct {
+	EntID uint32
```
[the remark this answers is not in the document]
Comment (pmazzini) [reply dhcpv6/option_vendorclass.go]: Sure

File: dhcpv6/option_vendorclass.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+			return nil, errors.New("ParseOptVendorClass: short data: missing length field")
+		}
+		vcLen := int(binary.BigEndian.Uint16(data[:2]))
+		if len(data) < vcLen+2 {
+			return nil, fmt.Errorf("ParseOptVendorClass: short data: less than %d bytes", vcLen+2)
+		}
+		opt.Data = append(opt.Data, data[2:vcLen+2])
+		data = data[2+vcLen:]
+	}
+	return &opt, nil
```
[the remark this answers is not in the document]
Comment (pmazzini) [reply dhcpv6/option_vendorclass.go]: The RFC doesn't specify that the vendor class data can't be empty.

[the remark this answers is not in the document]
Comment (insomniacslk) [reply dhcpv6/option_vendorclass.go]: > The
>   information contained in the data area of this option is contained in
>   one or more opaque fields that identify details of the hardware
>   configuration.

even if they don't use "MUST", they say "one or more"

**Pull request merged**